src/nimony/sem.nim

```diff
@@ -5763,22 +5764,33 @@ proc semPragmaLine(c: var SemContext; it: var Item; isPragmaBlock: bool) =
         inc it.n
 
     skipParRi it.n
-
-    if args.len != 2 and args.len != 3:
-      buildErr c, info, "build expected 2 or 3 parameters"
+    if args.len < 2:
+      buildErr c, info, "build expected at least 2 parameters"
 
     # XXX: Relative paths in makefile are relative to current working directory, not the location of the makefile.
-    let curWorkDir = os.getCurrentDir()
+    let curWorkDir = c.g.config.currentPath
     let currentDir = absoluteParentDir(info.getFile)
 
-    # Extract build pragma arguments
-    let compileType = args[0]
-    var name = replaceSubs(args[1], currentDir, c.g.config).toAbsolutePath(currentDir)
-    let customArgs = if args.len == 3: replaceSubs(args[2], currentDir, c.g.config) else: ""
-
-    if not semos.fileExists(name):
-      buildErr c, info, "cannot find: " & name
-    name = name.toRelativePath(curWorkDir)
+    var customIndex = 2
+    var name = default(string)
+    let compileType = args[0].toLowerAscii()
+    # Extract build pragma compile type
+    if compileType == "c" or compileType == "l":
```

Comment:
> {.compile.}, {.link.}, {.passC.} and {.passL.} pragmas could be implemented as a sugars of {.build.} pragma for compat with nim 2.x but it should be for another pull request

"c" meant backend types, i.e., c, cpp, rust etc. before

Reply:
so how could we fit the passC and passL and link?, maybe using `"c.passC"`. `"c.passL"`, `"c.link"` in the case for c backend for example?